modules/ubi/README.md

```diff
@@ -0,0 +1,59 @@
+# User Behavior Insights (UBI)
+
+UBI facilitates storing queries and events for the purposes of improving search relevance as descrbed by [[RFC] User Behavior Insights](https://github.com/opensearch-project/OpenSearch/issues/12084).
+
+## Indexing Queries
+
+For UBI to index a query, add a `ubi` block to the `ext` in the search request containing a `query_id`:
```

Comment:
I had it generate a `query_id` if none is provided but I should not have yet. In this first version, a `query_id` is required. This is because the search response is not yet being modified. In a later revision, `query_id` will be optional and generated if not provided and returned in the search response's `ext`. I will remove that code in `getQueryId()` to make a random UUID if it's `null`.

An empty block was sufficient, but I will change it to require that the `ubi` block contains a `query_id`. If no `ubi` block, or an empty `ubi` block, the rest of the code in the `UbiActionFilter` will be skipped.